frontend/src/features/LLM/MwasHypothesisGenerator.tsx

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+
+import { useLazyGetHypothesisQuery } from '../../api/client.ts';
+import { load } from '@fingerprintjs/botd';
+
+import IconButton from '@mui/material/IconButton';
+import BlurOnIcon from '@mui/icons-material/BlurOn';
+import Box from '@mui/material/Box';
+import Skeleton from '@mui/material/Skeleton';
+import Typography from '@mui/material/Typography';
+import Link from '@mui/material/Link';
+import Tooltip from '@mui/material/Tooltip';
+
+const MwasHypothesisGenerator = ({ identifiers, virusFamilies }) => {
+    const [getHypothesisText, { data: hypothesisData, isLoading: isLoadingHypothesis, error: errorHypothesis }] =
+        useLazyGetHypothesisQuery();
+
+    const onButtonClick = async () => {
+        try {
+            const botd = await load();
+            const detect = await botd.detect();
+            if (!detect || detect.bot === true) {
+                throw new Error('Bot detected');
+            }
+            getHypothesisText({
+                idColumn: 'bioproject',
+                ids: identifiers ? identifiers['bioproject'].single : [],
+                idRanges: identifiers ? identifiers['bioproject'].range : [],
+                virusFamilies: virusFamilies ? virusFamilies : [],
+                pageStart: 0,
+                pageEnd: 1000,
```

Comment:
these are only relevant for MWAS results, which can have a large number of rows. the hypothesis generator will only return one response at a time

Reply:
Removed parameter, see latest commit: d56d4bbb08682c5672d2ba3d502e6597fc476ab6